package-recipe.el

```diff
@@ -0,0 +1,108 @@
+;;; package-recipe.el --- Package recipes as EIEIO objects  -*- lexical-binding: t -*-
+
+;; Copyright (C) 2018  Jonas Bernoulli
+
+;; Author: Jonas Bernoulli <jonas@bernoul.li>
+
+;; This file is not (yet) part of GNU Emacs.
+;; However, it is distributed under the same license.
+
+;; GNU Emacs is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation; either version 3, or (at your option)
+;; any later version.
+
+;; GNU Emacs is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs; see the file COPYING.  If not, write to the
+;; Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+;; Boston, MA 02110-1301, USA.
+
+;;; Commentary:
+
+;; Package recipes as EIEIO objects.
+
+;;; Code:
+
+(require 'eieio)
+
+(declare-function package-build-recipe-alist "package-build" ())
+(defvar package-build-working-dir)
+
+;;; Classes
+
+(defclass package-recipe ()
+  ((url-format      :allocation :class       :initform nil)
+   (repopage-format :allocation :class       :initform nil)
+   (tag-regexp      :allocation :class       :initform nil)
+   (stable-p        :allocation :class       :initform nil)
+   (name            :initarg :name           :initform nil)
+   (url             :initarg :url            :initform nil)
+   (repo            :initarg :repo           :initform nil)
+   (repopage        :initarg :repopage       :initform nil)
+   (files           :initarg :files          :initform nil)
+   (branch          :initarg :branch         :initform nil)
+   (commit          :initarg :commit         :initform nil)
+   (version-regexp  :initarg :version-regexp :initform nil)
+   (old-names       :initarg :old-names      :initform nil))
+  :abstract t)
+
+(defun package-recipe-lookup (name)
+  (let ((plist (cdr (assq (intern name) (package-build-recipe-alist)))))
```

Comment:
I presume `plist` here is the actual recipe submitted by contributors? e.g., [`recipes/magit`](https://github.com/melpa/melpa/blob/master/recipes/magit)?